### ISSUE TITLE
can't acces internal values of dictionnary

### ISSUE THREAD

**Issue by huwilerb** [spyder-ide/spyder#14399]
### What steps will reproduce the problem?

In the Variable explorer, I try to navigate into a dictionnary. I get an internal problem. It is a nested dict, I can go in the dictionnary and see the outer keys, but not acces the inner key and values. The code itself can access the entire dict

### Traceback
```python-traceback
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionsdelegate.py", line 159, in createEditor
    editor.setup(value, key, icon=self.parent().windowIcon(),
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1308, in setup
    self.widget = CollectionsEditorWidget(self, self.data_copy,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1249, in __init__
    self.editor = CollectionsEditorTableView(self, data, readonly,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1115, in __init__
    self.source_model = CollectionsModelClass(self, data, title,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 143, in __init__
    self.set_data(data)
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 173, in set_data
    self.keys = sorted(list(data.keys()))
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionsdelegate.py", line 159, in createEditor
    editor.setup(value, key, icon=self.parent().windowIcon(),
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1308, in setup
    self.widget = CollectionsEditorWidget(self, self.data_copy,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1249, in __init__
    self.editor = CollectionsEditorTableView(self, data, readonly,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1115, in __init__
    self.source_model = CollectionsModelClass(self, data, title,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 143, in __init__
    self.set_data(data)
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 173, in set_data
    self.keys = sorted(list(data.keys()))
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionsdelegate.py", line 159, in createEditor
    editor.setup(value, key, icon=self.parent().windowIcon(),
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1308, in setup
    self.widget = CollectionsEditorWidget(self, self.data_copy,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1249, in __init__
    self.editor = CollectionsEditorTableView(self, data, readonly,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1115, in __init__
    self.source_model = CollectionsModelClass(self, data, title,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 143, in __init__
    self.set_data(data)
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 173, in set_data
    self.keys = sorted(list(data.keys()))
TypeError: '<' not supported between instances of 'str' and 'int'
Traceback (most recent call last):
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionsdelegate.py", line 159, in createEditor
    editor.setup(value, key, icon=self.parent().windowIcon(),
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1308, in setup
    self.widget = CollectionsEditorWidget(self, self.data_copy,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1249, in __init__
    self.editor = CollectionsEditorTableView(self, data, readonly,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 1115, in __init__
    self.source_model = CollectionsModelClass(self, data, title,
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 143, in __init__
    self.set_data(data)
  File "C:\Users\blaise.huwiler\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionseditor.py", line 173, in set_data
    self.keys = sorted(list(data.keys()))
TypeError: '<' not supported between instances of 'str' and 'int'
```

## Versions

* Spyder version: 4.1.4 
* Python version: 3.8.3
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.5.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.16.1 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
paramiko >=2.4.0               :  2.7.1 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.0 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=1.0                   :  2.5.3 (OK)
pyls >=0.34.0;<1.0.0           :  0.34.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.2 (OK)
qtconsole >=4.6.0              :  4.7.5 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.1.2 (OK)
spyder_kernels >=1.9.2;<1.10.0 :  1.9.2 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.1 (OK)

# Optional:
cython >=0.21                  :  0.29.21 (OK)
matplotlib >=2.0.0             :  3.2.2 (OK)
numpy >=1.7                    :  1.18.5 (OK)
pandas >=0.13.1                :  1.0.5 (OK)
scipy >=0.17.0                 :  1.5.0 (OK)
sympy >=0.7.3                  :  1.6.1 (OK)
```

**Comment by hengin** [comment]
This looks like the same problem as in #13481.

Try upgrading your Spyder version. This issue should be resolved in 4.2.0.

**Comment by ccordoba12** [comment]
Thanks @hengin for chiming in. I think you're right, @huwilerb's problem is a duplicate of issue #13481.

@huwilerb, unfortunately Spyder 4.2.0 is (still) not available in Anaconda (after more than one month of being released). However, you can give it a try by using our [Windows installer](https://github.com/spyder-ide/spyder/releases/download/v4.2.0/Spyder_64bit_full.exe).
